docs/lib/project-setup/fragments/android/coroutines/coroutines.md

```diff
@@ -0,0 +1,125 @@
+Amplify provides an optional and separate API surface which is entirely focused on using Kotlin's [coroutines](https://developer.android.com/kotlin/coroutines) and [flows](https://developer.android.com/kotlin/flow).
+
+To use it, import **`Amplify`** facade from `core-kotlin` instead of from `core`.
+
+With the Coroutines APIs, most Amplify functions are expressed as `suspend` functions:
+
+```kotlin
+import com.amplifyframework.kotlin.core.Amplify
+// ...
+
+val post = Post.builder()
+    .title("My First Post")
+    .build()
+
+lifecycleScope.launch {
+    try {
+        Amplify.DataStore.save(post) // This is suspending function!
+        Log.i("AmplifyKotlinDemo", "Saved a post")
+    } catch (failure: DataStoreException) {
+        Log.e("AmplifyKotlinDemo", "Save failed", failure)
+    }
+}
+```
+
+Compared to the traditional callback API, this doesn't make a big difference when used for a single method call.
+
+However, it greatly improves readability when chaining asynchronous calls. Moreover, you can use scopes, dispatchers, and other Kotlin coroutine primitives to get more control over your execution context.
+
+Let's revisit our nested example where we saved `Post`, `Editor`, and `PostEditor`. With Amplify's coroutines interface, we can write these operations sequentially:
+
+
+```kotlin
+lifecycleScope.launch {
+    try {
+        Amplify.DataStore.save(post)
+        Amplify.DataStore.save(editor)
+        Amplify.DataStore.save(postEditor)
+        Log.i("AmplifyKotlinDemo", "Post, Editor, and PostEditor saved")
+    } catch (failure: DataStoreException) {
+        Log.e("AmplifyKotlinDemo", "An item failed to save", failure)
+    }
+}
+```
+
+Compared to nesting these dependent calls in callbacks, this provides a much more readable pattern.
+
+## Installation
+
+Amplify's coroutine support is included in an optional module, `core-kotlin`. To start using the coroutine APIs, add the following dependency to your application's Gradle file:
+
+Under **Gradle Scripts**, open **build.gradle (Module: [YourApplicationName])**.
+
+Add the following line in `dependencies`:
+
+```groovy
+dependencies {
+    // Add the below line in `dependencies`
+    implementation 'com.amplifyframework:core-kotlin:0.1.0'
```

Comment:
```suggestion
    implementation 'com.amplifyframework:core-kotlin:0.1.1'
```

Reply:
Including this into an offline edit I'm about to push up! Thanks